heron/common/src/cpp/network/connection.cpp

```diff
@@ -62,25 +61,26 @@ Connection::~Connection() {
   delete mIncomingPacket;
   {
     for (auto iter = mOutstandingPackets.begin(); iter != mOutstandingPackets.end(); ++iter) {
-      delete iter->first;
+      delete *iter;
```

Comment:
instead of for - you could use while similar to the mSentPackets (that follows)?